### DISCUSSION TITLE
Lesson 4 issues with python terminal PLEASE HELP!!!!

### DISCUSSION THREAD

**Discussion by LesBLUUs** [smartcontractkit/full-blockchain-solidity-course-py#1440] [Q&A] [answered]
Okay so I have been following along just fine for the most part using VS code to complete each task. However, I downloaded Node.js(for a separate project not this one) along with addons that included a python setup tool or something. My issue now is that I cannot run the "python deploy.py" command successfully anymore. I tried reinstalling pip and the other suggested downloads to no avail. I noticed that my terminal changed from powershell to python also. I'm fairly new to all this did some googling but I can't seem to crack this alone. Any help would be greatly appreciated. Thank you in advance!!

ModuleNotFoundError: No module named 'pkg_resources'

That is the error I keep receiving btw.
![Screenshot 2022-04-23 lesson 4 error](https://user-images.githubusercontent.com/97270916/164934541-17c04c2e-8626-47f5-84fc-f75ee1b0a0a1.png)
![Screenshot 2022-04-23 error2](https://user-images.githubusercontent.com/97270916/164934545-786d709e-eb87-404b-8a7a-1badc01c978c.png)
The second image is the error I get when trying to install anything.

**Comment by cromewar** [comment, marked as answer]
Hello @LesBLUUs 
Python 3.10 has problems with web3 libs so, could you try again with pyton 3.8?
Additionally please consider to use WSL.

**Reply by LesBLUUs** [reply]
Okay that sounds great, thank you! 
is there a particular way that I should save the python file? I read something about making sure it's set to the correct PATH or something along those lines.
lastly, noob question but what is WSL and is there a particular way that I should use it upon my attempt? Thank you greatly your information is much appreciated!!!!

**Reply by cromewar** [reply]
You can select the path of python installation using the official python installer, (do not install python using microsft store)
WSL is Windows subsystem for linux you can find a entire guide on the course index: windows support appart.

**Reply by LesBLUUs** [reply]
Thanks for the tip. I had no luck adjusting my path correctly, so I ended up deleting python 3.10 and got an older version instead. I also went ahead and got started with WSL. It seems more complicated honestly lol. I'm a little confused so would I be using this instead of the terminal inside vs code?

If so am I to reinstall the pip and solcx and so on in the wsl environment? I know it's that I likely sound crazy but i'm a little out of sorts here.

**Reply by LesBLUUs** [reply]
Also I installed WSL 1 is WSL 2 recommended instead?
I found that there is a plug-in to use WSL and I have downloaded that btw

**Reply by cromewar** [reply]
Well WSL is for installing a linux instance of your computer which is independent from the windows installation, pelase give a check the article on "windows support" apart on the course index.
